test/system/distr_tests.cc

```diff
@@ -371,5 +372,66 @@ namespace detail {
 		CHECK(host_rank == global_rank);
 	}
 
+	TEST_CASE_METHOD(test_utils::runtime_fixture, "command graph can be collected across distributed nodes", "[print_graph]") {
+		env::scoped_test_environment tenv(recording_enabled_env_setting);
+
+		int global_size = 0;
+		MPI_Comm_size(MPI_COMM_WORLD, &global_size);
+		if(global_size != 2) { SKIP("can only perform this test when invoked for exactly 2 participating nodes"); }
+
+		distr_queue q;
+		celerity::range<2> range(16, 16);
+		celerity::buffer<float, 2> buff_a(range);
+
+		// set small horizon step size so that we do not need to generate a very large graph to test this functionality
+		auto& tm = celerity::detail::runtime::get_instance().get_task_manager();
+		tm.set_horizon_step(1);
+
+		for(int i = 0; i < 2; ++i) {
+			q.submit([&](handler& cgh) {
+				celerity::accessor acc_a{buff_a, cgh, celerity::access::one_to_one{}, celerity::read_write};
+				cgh.parallel_for<class UKN(cluster_graph_printing)>(range, [=](item<2> item) { (void)acc_a; });
+			});
+		}
+
+		q.slow_full_sync();
+
+		const auto graph = celerity::detail::runtime::get_instance().gather_command_graph();
+
+		// only check on master node, since that is where the graph is gathered
+		int global_rank = 0;
+		MPI_Comm_rank(MPI_COMM_WORLD, &global_rank);
+		if(global_rank == 0) {
+			// Smoke test: It is valid for the dot output to change with updates to graph generation. If this test fails, verify that the printed graphs are
+			// sane and complete, and if so, replace the `expected` values with the new dot graph.
+			const std::string expected =
```

Comment:
Since these are annoying to update, couldn't we just check for something like `StringContains("<b>execution</b> [[0,0,0] - [8,16,1]]") && StringContains("<b>execution</b> [[8,0,0] - [16,16,1]]")`?

Reply:
Since this is the only test that actually checks the graph composition from multiple sources, I'd like to keep it complete.